nixos/modules/services/networking/wstunnel.nix

```diff
@@ -0,0 +1,341 @@
+        type = types.port;
+      };
+
+      autoStart = mkOption {
+        description = mdDoc "Whether this tunnel server should be started automatically.";
+        type = types.bool;
+        default = true;
+      };
+
+      restrictTo = mkOption {
+        description = mdDoc "Accepted traffic will be forwarded only to this service. Set to `null` to disable this option.";
+        type = types.nullOr types.str;
+        example = "127.0.0.1:51820";
+      };
+
+      enableHTTPS = mkOption {
+        description = mdDoc "Use HTTPS for the tunnel server.";
+        type = types.bool;
+        default = false;
+      };
+
+      tlsCertificate = mkOption {
+        description = mdDoc "TLS certificate to use instead of the hardcoded one in case of HTTPS connections. Use together with `tlsKey`.";
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/acme/example.com/cert.pem";
+      };
+
+      tlsKey = mkOption {
+        description = mdDoc "TLS key to use instead of the hardcoded on in case of HTTPS connections. Use together with `tlsCertificate`.";
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/acme/example.com/key.pem";
+      };
+
+      verboseLogging = mkOption {
+        description = mdDoc "Enable verbose logging.";
+        type = types.bool;
+        default = false;
+      };
+
+      runAsRoot = mkOption {
+        description = mdDoc "Run `wstunnel` as root.";
+        type = types.bool;
+        default = false;
+      };
+
+      package = mkOption {
+        description = mdDoc "The `wstunnel` package to use for this instance.";
+        type = types.package;
+        default = cfg.package;
+        defaultText = literalExpression "services.wstunnel.package";
+      };
+
+      extraArgs = mkOption {
+        description = mdDoc "Extra command line arguments to pass to `wstunnel`.";
+        type = types.str;
+        default = "";
+      };
+    };
+  };
+  clientOptions = {
+    options = {
+      address = mkOption {
+        description = mdDoc "Server address to connect to.";
+        type = types.str;
+        example = "example.com";
+      };
+
+      port = mkOption {
+        description = mdDoc "Server port to connect to.";
+        type = types.port;
+      };
+
+      enableHTTPS = mkOption {
+        description = mdDoc "Enable HTTPS when connecting to the server.";
+        type = types.bool;
+        default = false;
+      };
+
+      autoStart = mkOption {
+        description = mdDoc "Whether this tunnel server should be started automatically.";
+        type = types.bool;
+        default = true;
+      };
+
+      localToRemote = mkOption {
+        description = mdDoc "Hosts and ports to listen on, plus the remote hosts and ports to forward traffic to.";
+        type = types.listOf types.str;
+        default = [];
+        example = [ "127.0.0.1:8080:127.0.0.1:8080" ];
```

Comment:
What is the host and what the local part?

Reply:
Clarified this in the doc